### DISCUSSION TITLE
Target Processor drop down list is missing with M55

### DISCUSSION THREAD

**Discussion by ronenw7** [eclipse-embed-cdt/eclipse-plugins#548] [Debug Plug-ins Q&A]
If you see GCC documentation that ARM shared ; you can see it allows to tune code for cortex M55. [https://gcc.gnu.org/onlined...](https://disq.us/url?url=https%3A%2F%2Fgcc.gnu.org%2Fonlinedocs%2Fgcc%2FARM-Options.html%3Ay4wDnJEcVsq0Ums-57ccMDdPsWE&cuid=6339073)

However I’ve noticed that M55 is missing from the "Target Processor" drop down list.
![TargetProcessor](https://user-images.githubusercontent.com/32730648/211797778-a533f14b-51e6-4d17-a3fe-90916fdb26c4.jpg)

**Comment by ilg-ul** [comment]
I will shortly upgrade the toolchain to the latest Arm sources and then update the plug-ins to match the list of supported devices.

Till then you can select **Toolchain default** and enter your choice in the **Other target flags** field.

**Reply by ronenw7** [reply]
Thanks.
What shall be the content for the **Other target flags**? ("cortex-m55")

**Reply by ilg-ul** [reply]
Any legal flags that you can pass to the compiler, as per manual.

**Reply by ronenw7** [reply]
I got it thanks,

1. Still, compiler yelled about:
arm-none-eabi-gcc: error: unrecognized -mcpu target: cortex-m5
![image](https://user-images.githubusercontent.com/32730648/211813769-961873e6-7f10-49af-a7b0-e6e55bd2d6e9.png)
Here is my settings:
![image](https://user-images.githubusercontent.com/32730648/211813561-831424d8-c9d1-4a7a-805a-b19f6515a848.png)


2. How I can setback the "Target Processor" to default values, would it be via "Restore Defaults" button?

**Reply by ilg-ul** [reply]
cortex-m5?

**Reply by ronenw7** [reply]
sorry - this is a typo error.
Pls see below failure:
![image](https://user-images.githubusercontent.com/32730648/211815959-3d58f01a-0254-4df1-bc7c-7f44c6939fd1.png)

**Reply by ronenw7** [reply]
I thinks i will just wait for you to "update the plug-ins".
Thanks a lot for your support for now.

**Reply by ilg-ul** [reply]
The definition in `DEVICE.h` file are incorrect.

**Reply by ilg-ul** [reply]
The plug-ins update will not solve your problem, you do not have a cortex-m55 valid project.

**Reply by ronenw7** [reply]
This is the Project i choose:
![image](https://user-images.githubusercontent.com/32730648/211821482-a967c522-c4b9-413f-80d3-d0e4baa531d9.png)
Is it wrong?

**Reply by ilg-ul** [reply]
It is not wrong, but it is incomplete, you have to provide device specific files, the project template has no idea what device you are using.

**Reply by gekakud** [reply]
Can you please guide us on how to do that or show us any references? Many thanks in advance!

**Reply by ilg-ul** [reply]
- <https://eclipse-embed-cdt.github.io/templates/cortexm/>

However please note that this template is very old and quite basic. A bit more elaborate are the STM32F templates, and the most advanced are the xPack templates.

But none will write the application in your place, and you have to get used to device specifics.

**Reply by hnhoan** [reply]
Can you tell me will the M55 be supported ? I use Eclipse embedded CDT and love it.  Would love to see more MCU supports added. Thanks so for this.

**Comment by ilg-ul** [comment]
Time permitting, there might be an update, which will address this issue too, following the next Arm GCC release.

Till then you can select **Toolchain default** and enter your choice in the **Other target flags** field.
